src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -129,19 +138,62 @@ public void run(ActionContext context) throws Exception {
 
     QueryJobConfiguration queryConfig = builder.build();
 
-    Job queryJob = bigQuery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
+    // Exponential backoff with initial retry of 1 second and max retry of 32 seconds.
+    executeQueryWithExponentialBackoff(bigQuery, queryConfig, context);
+  }
 
-    LOG.info("Executing SQL as job {}.", jobId.getJob());
-    LOG.debug("The BigQuery SQL is {}", config.getSql());
+  protected void executeQueryWithExponentialBackoff(BigQuery bigQuery,
+                                                   QueryJobConfiguration queryConfig, ActionContext context) {
+    Failsafe.with(getRetryPolicy()).run(() -> executeQuery(bigQuery, queryConfig, context, retryOnReason));
+  }
 
-    // Wait for the query to complete
-    queryJob = queryJob.waitFor();
+  private RetryPolicy<Object> getRetryPolicy() {
+    // Exponential backoff with initial retry of 1 second and max retry of 32 seconds.
+    Duration initialRetryDuration = Duration.ofSeconds(1);
+    Duration maxRetryDuration = Duration.ofSeconds(32);
+    int multiplier = 2;
+    int maxRetryCount = 5;
```

Comment:
All of this should be configurable. Retries must be disableable

Reply:
Agree. Please add a `Retry Configuration` section in the UI and allow the users to fine-tune this behavior.

Reply:
@tivv @fernst It was mentioned previously on the bug ticket also that Bigquery SLA is for 32 seconds and that is the reason we have added these fix properties to go upto that limit. Same was confirmed in design doc also.
Can you please confirm in design doc also. Shared with you by puspender.

Reply:
Few points here:

-  Customer may not wish to wait so long. Yep, BQ may recover, but it's also possible that it's too long for given customer
- You are changing the behaviour. We must have an option to disable this behavior
- SLAs are statistical. They may still be violated
- SLAs are point in time. They may change and we need to be able to adjust the bahvior without code change